src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,42 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final int DATE_INDEX = 0;
+    private static final int NAME_INDEX = 1;
+    private static final int WORKING_HOURS_INDEX = 2;
+    private static final int INCOME_PER_HOURS_INDEX = 3;
+    private static final String HEADER = "Report for period ";
+    private static final String SEPARATOR = " - ";
+    private static final String DATA_SEPARATOR = " ";
+    private DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder result = new StringBuilder();
+        result.append(HEADER).append(dateFrom).append(SEPARATOR).append(dateTo);
+        LocalDate localDateFrom = LocalDate.parse(dateFrom, formatter);
+        LocalDate localDateTo = LocalDate.parse(dateTo, formatter);
+        for (String name : names) {
+            result.append("\n");
```

Comment:
Please use operating system independent symbol for new line
```suggestion
            result.append(System.lineSeparator());
```